Dockerfile.cpu

```diff
@@ -1,24 +1,32 @@
-FROM ubuntu:18.04
+FROM --platform=$TARGETPLATFORM ubuntu:20.04
 MAINTAINER Roboflow <docker@roboflow.com>
 ENV TZ=Etc/UTC
 ENV ROBOFLOW_PACKAGE="roboflow-node"
 
-RUN apt update
-RUN apt-get -y install curl net-tools build-essential python3
+RUN apt update && \
+    apt-get -y install curl net-tools build-essential python3
 
 # Install NVM and Node.js LTS
-RUN curl -L https://raw.githubusercontent.com/tj/n/master/bin/n -o /usr/local/bin/n
-RUN chmod +x /usr/local/bin/n
-RUN /usr/local/bin/n lts
-
-RUN npm install pm2 -g
+RUN curl -L https://raw.githubusercontent.com/tj/n/master/bin/n -o /usr/local/bin/n && \
+    chmod +x /usr/local/bin/n && \
+    /usr/local/bin/n lts
 
 COPY ["./", "/inference-server"]
 RUN cp /inference-server/server/configs/package.cpu.json /inference-server/server/package.json
 
+ARG TARGETPLATFORM
+ARG TARGETOS
+ARG TARGETARCH
+
 # Install proper dependencies
 WORKDIR /inference-server/server
-RUN rm -rf ./node_modules
-RUN npm install
+RUN rm -rf ./node_modules && \
+    npm install --target_platform=$TARGETOS --target_arch=$TARGETARCH && \
+    npm install --target_platform=$TARGETOS --target_arch=$TARGETARCH pm2 -g --verbose
+
+# tfjs on arm needs to be rebuilt from source
+RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then \
```

Comment:
Looks awesome - curious where you learned this pattern? Common node docker cross-build practice?

Reply:
Didn’t seem common, no. I kind of hacked it together by combining a few different things I found.